cli/pcluster/cw-dashboard-substack.cfn.yaml

```diff
@@ -0,0 +1,253 @@
+                {%- endfor %}
+                {{ update_position(6,6) }}
+              {%- endfor %}
+
+              {# Conditional EBS metrics -#}
+              {%- set ebs_metrics_conditions = [{'metric': 'VolumeConsumedReadWriteOps', 'accepted_vol_types': ["io1"]},
+                                                {'metric': 'VolumeThroughputPercentage', 'accepted_vol_types': ["io1"]},
+                                                {'metric': 'BurstBalance', 'accepted_vol_types': ["gp2", "st1", "sc1"]}]
+              %}
+              {%- for metric_condition in ebs_metrics_conditions %}
+                {%- set isPresent = {'bool': False} %}
+                {%- for i in range(number_of_ebs_volumes) if not isPresent.bool %}
+                  {%- if ebs_volume_types[i] in metric_condition.accepted_vol_types %} {# we should break, needs Loop Controls extension #}
+                    {%- if isPresent.update({'bool': True}) %} {% endif %}
+                  {%- endif %}
+                {%- endfor %}
+                {%- if isPresent.bool %}
+              # EBS {{ metric_condition.metric }} graph
+              - !Sub ',{"type":"metric","x":{{ coord.x }},"y":{{ coord.y }},"width":6,"height":6,"properties":{"view":"timeSeries","stacked":false,"region":"${AWS::Region}","metrics":'
+                  {%- set rendered_metrics_group_list = [] %} {#- FIXME This would be interesting so that we could join using commas. -#}
+                  {#- Previous line would also be useful to concatenate the lines for the different volumes together (but would it be better? -#}
+                  {#- only one Sub for 5 variables but all characters for the different volumes would be concatenated... maybe less readable?) -#}
+                  {%- set first_metric = {'bool': True} %}
+                  {%- for i in range(number_of_ebs_volumes) %}
+                    {%- if ebs_volume_types[i] in metric_condition.accepted_vol_types %}
+              - !Sub
+                - '{% if first_metric.bool %}{{ first_metric.update({'bool': False}) or "" }}[{% else %},{% endif %}["AWS/EBS","{{ metric_condition.metric }}","VolumeId","${EBS_Volume{{ i|int +1 }}}"]'
+                - EBS_Volume{{ i|int +1 }}: !Select
+                    - '{{ i|int }}'
+                    - !Ref 'EBSVolumesIDs'
+                    {%- endif %}
+                  {%- endfor %}
+                  {{ update_position(6,6) }}
+              - ']}}'
+                {% endif %} {# if isPresent.bool #}
+              {%- endfor %}
+              {{ update_pos_after_section(6) }}
+
+              {#- Unconditional RAID metrics #}
+              # RAID metrics graphs
+              {%- set number_of_raid_volumes = config.RAIDOptions.split(',')[2]|int -%} {# if NONE, converts to 0! #}
+              {%- if number_of_raid_volumes > 0 %}
+              - !Join # TODO: remove this join that is not needed anymore since we do not use the CreateRAIDSubstack condition
```

Comment:
There are a lot of TODO and FIXME, let's double check and remove them.

Reply:
Done, still a TODO but this is to enable 'break' inside jinja, might not be needed